### PR TITLE
Reimplement CI on GHA

### PR DESCRIPTION
**Pull request by therealprof**
Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

**Comment by ZeroErrors** [comment]
Maybe could add a cache for the downloaded dependencies to reduce build time even more.
I haven't had to use the cache action before but I think this is the correct config:
```yaml
- name: Cache cargo registry
  uses: actions/cache@v2
  with:
    path: ~/.cargo/registry
    key: ${{ matrix.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
```

**Comment by therealprof** [comment]
@ZeroErrors Yeah, let's try to get it running first before we optimize it. I just had some fun with trying to use env variables; seems I cannot get them to work in matrix definitions... 🤷🏻‍♂️😅

**Comment by therealprof** [comment]
r? @adamgreig 

I think this should be good for now.

**Comment by ZeroErrors** [comment]
Yeah for sure, just thought I'd suggest it anyway.

Was just digging into why it wasn't working since it seems logical that it should and from the looks of the documentation its very limited where you can use `${{ env.VAR }}`
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#env-context
> You can only use the `env` context in the value of the `with` and `name` keys, or in a step's `if` conditional. For more information on the step syntax

**Comment by therealprof** [comment]
The documentation on GHA is still quite lackluster, I didn't even stumble across the linked page when researching.

File: .github/bors.toml [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
 delete_merged_branches = true
 required_approvals = 1
-status = ["continuous-integration/travis-ci/push"]
+status = [
+        "ci (1.38.0, ubuntu-latest, false)",
+        "ci (1.38.0, macos-latest, false)",
+        "ci (1.38.0, windows-latest, false)",
+        "ci (stable, ubuntu-latest, false)",
+        "ci (stable, macos-latest, false)",
+        "ci (stable, windows-latest, false)",
```
Comment (adamgreig) [inline .github/bors.toml:10]: Do we want rustfmt here too?

Comment (therealprof) [reply .github/bors.toml:10]: Later, maybe; I didn't want to do a lot more than Travis is currently doing now in case we want to take it back.

**Pull request merged**